lib/VisualGenerator.js

```diff
@@ -364,6 +361,35 @@ class VisualGenerator {
         }
         return true;
     }
+
+    /**
+     * Check visual name 
+     * Using https://github.com/mathiasbynens/mothereff.in/tree/master/js-properties
+     * 
+     * @static
+     * @param {string} name Visual name
+     * @returns {boolean} is valid?
+     * 
+     * @memberof VisualGenerator
+     */
+    static checkVisualName(name) {
+        const regexES3ReservedWord = /^(?:do|if|in|for|int|new|try|var|byte|case|char|else|enum|goto|long|null|this|true|void|with|break|catch|class|const|false|final|float|short|super|throw|while|delete|double|export|import|native|public|return|static|switch|throws|typeof|boolean|default|extends|finally|package|private|abstract|continue|debugger|function|volatile|interface|protected|transient|implements|instanceof|synchronized)$/;
+        const regexNumber = /^(?![+-])([0-9\+\-\.]+)/;
+        const regexZeroWidth = /\u200c|\u200d/;
+        const valueAsUnescapedString = name.replace(/\\u([a-fA-F0-9]{4})|\\u\{([0-9a-fA-F]{1,})\}/g, ($0, $1, $2) => {
+            const codePoint = parseInt($2 || $1, 16);
+            // If it’s a surrogate…
+            if (codePoint >= 0xD800 && codePoint <= 0xDFFF) {
+                // Return a character that is never valid in an identifier.
+                // This prevents the surrogate from pairing with another.
+                return '\0';
+            }
+            return String.fromCodePoint(codePoint);
+        });
+        return !regexNumber.test(name) &&
```

Comment:
what error is shown to the user if he uses a bad name?
We should be specific in letting the user know what name convention to use so he can correct it.
Either in documentation of the tool or in the actual error returned: for example “ ’function’ is not a valid word within a visual name”